sktree/tree/_unsup_criterion.pxd

```diff
@@ -53,4 +53,11 @@ cdef class UnsupervisedCriterion(BaseCriterion):
     cdef void init_feature_vec(
         self,
         const DTYPE_t[:] Xf,
+    ) nogil
+
+    cdef double sum_of_squares(
```

Comment:
Generic unsupervisedcriterions don't need to necessarily compute sum of squares (I.e. variance), so this shouldn't be in the base class.